### PR TITLE
shell: Update exit code for special cases

### PR DESCRIPTION
**Pull request by chu11**
Per discussion in #2752, quick fix to handle special case exit codes in the job shell.

### CODE REVIEW

File: src/shell/shell.c [hunk 1]
```diff
@@ -1114,7 +1114,17 @@ int main (int argc, char *argv[])
...
-            shell_die (1, "task %d: start failed: %s: %s",
+            int ec = 1;
+            /* bash standard, 126 for permission/access denied, 127 for
+             * command not found.  68 (EX_NOHOST) for No route to host.
+             */
+            if (errno == EPERM || errno == EACCES)
+                ec = 126;
+            else if (errno == ENOENT)
+                ec = 127;
+            else if (errno == EHOSTUNREACH)
```
Comment (grondo) [inline src/shell/shell.c]: Since the shell is only launching local tasks, can we ever get `EHOSTUNREACH` from shell_task_start?
(I'm probably just missing a corner case, since you probably saw this errno during testing)

File: t/t2608-job-shell-log.t [hunk 2]
```diff
@@ -133,12 +133,20 @@ for topic in "shell.init" "shell.exit" \
...
 	grep "job.exception type=exec severity=0 task 0: start failed" missing.err &&
         grep "No such file or directory" missing.err
 '
 
+test_expect_success 'flux-shell: illegal command logs fatal error' '
+        mkdir adirectory &&
+	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
+	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
+	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
+        grep "Permission denied" illegal.err
```
Comment (grondo) [inline t/t2608-job-shell-log.t]: nit: looks like maybe a couple indentation errors in this test?

File: src/shell/shell.c (hunk 1 above)
Comment (chu11) [reply src/shell/shell.c]: > (I'm probably just missing a corner case, since you probably saw this errno during testing)

You think too highly of me :-)  I blindly copied behavior from other parts of Flux.  You're absolutely right, the shell only does local tasks.

File: t/t2608-job-shell-log.t (hunk 2 above)
Comment (chu11) [reply t/t2608-job-shell-log.t]: Yup, there was a tab vs space issue.  Went through the file and it seems we got a lot of tab vs space inconsistencies, and in other tests.  Maybe should do a big cleanup one of these days.

**Comment by chu11** [comment]
re-pushed, removing EHOSTUNREACH and tweaking some tabbing per comments above.  There are more tabbing inconsistencies in `t2608-job-shell-log.t`, but those fixes are probably for another PR.

**Comment by grondo** [comment]
>  There are more tabbing inconsistencies in t2608-job-shell-log.t, but those fixes are probably for another PR.

Yeah, sorry that must be my fault...

**Comment by chu11** [comment]
> Yeah, sorry that must be my fault...

Skimming through the tests, the inconsistencies are everywhere, so I'm sure it was a collective effort! :-)

**Comment by chu11** [comment]
Hmmm, almost all builders failed:

```
expecting success: 
	mkdir adirectory &&
	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
	grep "Permission denied" illegal.err

not ok 132 - flux-shell: illegal command logs fatal error
```

Nothing on stderr, so mkdir failed?  Perhaps I need to `chdir` to temp or something for travis.

Removing merge-when-passing so I can add some debugging commits

**Comment by chu11** [comment]
hmmm

```
expecting success: 
	mkdir adirectory &&
flux mini run adirectory &&
	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
	grep "Permission denied" illegal.err

0.076s: flux-shell[0]: FATAL: task 0: start failed: adirectory: No such file or directory
0.077s: job.exception type=exec severity=0 task 0: start failed: adirectory: No such file or directory
flux-job: task(s) exited with exit code 127
not ok 132 - flux-shell: illegal command logs fatal error
```

I'm getting "No such file or directory", but the `mkdir adirectory` clearly succeeds on the line above.  `umask` issue? hmmm

**Comment by chu11** [comment]
Guess it was a PATH issue, this fixes things:

```
-       test_expect_code 126 flux mini run adirectory 2>illegal.err &&
+       test_expect_code 126 flux mini run ./adirectory 2>illegal.err &&
```

squashed, rebased, and re-pushed.  I don't think I should reset `merge-when-passing`, so if someone else can, thanks!

**Comment by grondo** [comment]
One quick thing -- the commit message still mentions EHOSTUNREACH

**Comment by chu11** [comment]
> One quick thing -- the commit message still mentions EHOSTUNREACH

Doh! Thanks, repushed

**Comment by grondo** [comment]
Nice! Re-setting merge-when-passing.
